systemtest/test_content_length.py

```diff
@@ -65,20 +65,23 @@ def test_content_length_0(self, posturl, crash_generator):
         )
 
     def test_content_length_20(self, posturl, crash_generator):
-        """Post a crash with a content-length 20"""
+        """Post a crash with a content-length 20 which is less than content"""
         raw_crash, dumps = crash_generator.generate()
 
         # Generate the payload and headers for a crash with no dumps
         payload, headers = mini_poster.multipart_encode(raw_crash)
 
+        assert int(headers["Content-Length"]) > 20
+
         # Add wrong content-length
         headers["Content-Length"] = "20"
 
         resp = http_post(posturl, headers, payload)
 
         assert resp.getcode() == 400
         assert (
-            str(resp.read(), encoding="utf-8") == "Discarded=malformed_no_annotations"
+            str(resp.read(), encoding="utf-8")
+            == "Discarded=malformed_invalid_payload_structure"
```

Comment:
The discarded messages changed a while ago in 29b5a831e86f4df480ed3adc10ade93638485987 (March 2022). I must not have run the systemtests since then. :(